spec/services/order_builder_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+RSpec.describe PurchasesHandler::OrderBuilder, type: :service do
+
+  describe "#call" do
+
+    context "when order is created" do
+      let(:user) { create(:user) }
+
+      it "builds order" do
+        expect(PurchasesHandler::OrderBuilder.call(user, {})).to be_valid  
```

Comment:
```suggestion
        order = PurchasesHandler::OrderBuilder.call(user, {})
        
        expect(order).to be_kind_of(Order)  
```

De uma lida nesse artigo: https://thoughtbot.com/blog/four-phase-test
Fica mais facil de entender quando você organiza o teste em 3 fases (setup, exercise, verify)